reactors/1D_packbed.ipynb

```diff
@@ -0,0 +1,518 @@
+    "\n",
+    "if membrane_present == True:\n",
+    "    print(\"Modeling packed-bed, catalytic-membrane reactor...\")\n",
+    "    print(membrane_sp_name, \"permeable membrane is present.\")\n",
+    "\n",
+    "# Get required properties based on the geometry and mechanism\n",
+    "W_g = gas.molecular_weights  # vector of molecular weight of gas species\n",
+    "vol_ratio = phi / (1 - phi)\n",
+    "eff_factor = phi / tau  # effective factor for permeability calculation\n",
+    "B_g = (\n",
+    "    pow(vol_ratio, 2) * pow(dp, 2) * eff_factor / 72\n",
+    ")  # permeability based on Kozeny-Carman equation\n",
+    "area2vol = 2 / R  # area to volume ratio assuming a cylindrical reactor\n",
+    "D_h = 2 * R  # hydraulic diameter\n",
+    "membrane_sp_ind = gas.species_index(\n",
+    "    membrane_sp_name\n",
+    ")  # index of the membrane-permeable species\n",
+    "\n",
+    "# Inlet operating conditions\n",
+    "T_in = 673  # inlet temperature [K]\n",
+    "p_in = 5e5  # inlet pressure [Pa]\n",
+    "v_in = 0.001  # inlet velocity [m/s]\n",
+    "T_wall = 723  # wall temperature [K]\n",
+    "h_coeff = 1e2  # heat transfer coefficient\n",
+    "\n",
+    "# Set gas and surface states\n",
+    "gas.TPX = T_in, p_in, \"NH3:0.99, AR:0.01\"  # inlet composition\n",
+    "surf.TP = T_in, p_in\n",
+    "Yk_0 = gas.Y\n",
+    "rhou0 = gas.density * v_in\n",
+    "\n",
+    "# Initial surface coverages\n",
+    "surf.advance_coverages(\n",
+    "    1e10\n",
+    ")  # advancing coverages over a long period of time to get the steady state.\n",
+    "Zk_0 = surf.coverages"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define residual function required for IDA solver"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def residual(z, y, yPrime, res):\n",
+    "    \"\"\"Solution vector for the model\n",
+    "    y = [rho*u, p, T, Yk, Zk]\n",
+    "    yPrime = [d(rho*u)dz, dpdz, dTdz, dYkdz, dZkdz]\n",
+    "    \"\"\"\n",
+    "    # Get current thermodynamic state from solution vector and save it to local variables.\n",
+    "    rhou = y[offset_rhou]  # mass flux (density * velocity)\n",
+    "    Y = y[offset_Y : offset_Y + n_gas]  # vector of mass fractions of all gas species\n",
+    "    Z = y[\n",
+    "        offset_Z : offset_Z + n_surf\n",
+    "    ]  # vector of site fractions of all surface species\n",
+    "    p = y[offset_p]  # pressure\n",
+    "    T = y[offset_T]  # temperature\n",
+    "\n",
+    "    drhoudz = yPrime[offset_rhou]  # mass-flux spatial derivative\n",
+    "    dYdz = yPrime[offset_Y : offset_Y + n_gas]  # mass fraction spatial derivative\n",
+    "    dZdz = yPrime[offset_Z : offset_Z + n_surf]  # site fraction spatial derivative\n",
+    "    dpdz = yPrime[offset_p]  # pressure spatial derivative\n",
+    "    dTdz = yPrime[offset_T]  # temperature spatial derivative\n",
```

Comment:
I don't think you need the end-of-line comments on any of these lines... it wouldn't matter, except the line break for `Z`...

Reply:
Updated.